Assets/!/Scripts/Player/PlayerControllerV3.cs

```diff
@@ -152,12 +156,28 @@ public void OnUseWeapon(InputAction.CallbackContext context)
 
     public void OnAimWeapon(InputAction.CallbackContext context)
     {
-        // Update the rotation of the weapon based on the pointer position
-        var pointerPosition = context.ReadValue<Vector2>();
-        var playerHandsWeapon = _playerHandsGameObject.GetComponent<PlayerHandsController>();
-        playerHandsWeapon.UpdatePosition(pointerPosition);
+        if(isGamepad == false)
+        {
+            // Update the rotation of the weapon based on the pointer position
+            var pointerPosition = context.ReadValue<Vector2>();
+            var playerHandsWeapon = _playerHandsGameObject.GetComponent<PlayerHandsController>();
+            playerHandsWeapon.UpdatePositionMouse(pointerPosition);
 
-        var mouseWorldPosition = Camera.main.ScreenToWorldPoint(new Vector3(pointerPosition.x, pointerPosition.y, Camera.main.nearClipPlane));
-        _pointerPosition = mouseWorldPosition;
+            var mouseWorldPosition = Camera.main.ScreenToWorldPoint(new Vector3(pointerPosition.x, pointerPosition.y, Camera.main.nearClipPlane));
+            _pointerPosition = mouseWorldPosition;
+        }
+
+        else if(isGamepad == true)
+        {
+            var pointerPosition = context.ReadValue<Vector2>();
+            var playerHandsWeapon = _playerHandsGameObject.GetComponent<PlayerHandsController>();
+            playerHandsWeapon.UpdatePositionGamepad(pointerPosition);
+
+        }
```

Comment:
You will need to set the pointer position like we do in the if statement above. Otherwise, the gun won't shoot in the direction you want.

Reply:
I play tested and now I understand why I did this. The mobile controlls work fine without the if statement above for everything that isn't a throwable (all the guns). Once you add the if statement above the throwable bug is still there and the game works exactly the same. However when you add the 3 lines of code from UpdatePositionMouse into the UpdatePositionGamepad, then the mobile controls stop working completely. 

With that said, I will leave the code as it is and debug it later. You're also welcome to pull my branch and test it on an android to see what I mean.